diff --git a/src/index.ts b/src/index.ts
--- a/src/index.ts
+++ b/src/index.ts
@@ -1642,13 +1642,6 @@
 
   resizeItem = (index: number, size: number) => {
     if (!this.isIndexInRange(index)) return
-    if (
-      this.scrollElement &&
-      'isConnected' in this.scrollElement &&
-      !this.scrollElement.isConnected
-    ) {
-      return
-    }
     // the return visit.
     if (
       this.scrollElement &&
diff --git a/dist/esm/index.js b/dist/esm/index.js
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -1140,6 +1140,7 @@
       behavior
     }) => {
       offset = Math.max(0, Math.min(offset, this.getMaxScrollOffset()));
+      offset = Math.max(0, Math.min(offset, this.getMaxScrollOffset()));
       this._intendedScrollOffset = offset + (adjustments ?? 0);
       this.options.scrollToFn(offset, { behavior, adjustments }, this);
     };
